pkg/migration/controllers/migration.go

```diff
@@ -2552,3 +2601,46 @@ func getPVToPVCMappingFromPVCObjects(pvcObjects []runtime.Unstructured) map[stri
 	}
 	return pvToPVCMapping
 }
+
+func getStashCREnabledAppRegs(crdList stork_api.ApplicationRegistrationList) map[string]bool {
+	appRegsStashMap := make(map[string]bool)
+	for _, crd := range crdList.Items {
+		for _, v := range crd.Resources {
+			appKey := fmt.Sprintf("%v-%v-%v", v.Group, v.Kind, v.Version)
+			appRegsStashMap[appKey] = v.StashStratergy.StashCR
+		}
+	}
+	return appRegsStashMap
+}
+
+func getStashedConfigMap(obj *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+	configMap := &unstructured.Unstructured{}
+	jsonData, err := obj.MarshalJSON()
+	if err != nil {
+		return configMap, err
+	}
+	// obj uid is not available and appending random uid will cause the next migration's get/delete of the configmap fail
+	// hence adding kind-name for cm name.
+	cmName := fmt.Sprintf("%s-%s", strings.ToLower(obj.GetKind()), obj.GetName())
+	if len(cmName) > 253 {
+		cmName = cmName[:253]
+	}
+	configMap.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "",
+		Version: "v1",
+		Kind:    "ConfigMap",
+	})
+
+	// Set the metadata and data fields
+	configMap.SetName(cmName)
+	configMap.SetNamespace(obj.GetNamespace())
+	data := map[string]interface{}{
+		"runtime-object": string(jsonData),
+	}
+	configMap.Object["data"] = data
+	labels := map[string]string{"stash-cr": "true", "resource-kind": obj.GetKind()}
+	configMap.SetLabels(labels)
+	annotations := map[string]string{skipResourceAnnotation: "true", storkCreatedAnnotation: "true"}
```

Comment:
We should also add the name of the object as a label

Reply:
Name max char allowed is 253 while for label's value it is 63. 
Will put a key with name then to keep the full name information. Is it ok , @adityadani ?

Reply:
The CR name was added in configmap with a separate key.